pkg/gardenadm/cmd/init/init.go

```diff
@@ -259,12 +259,24 @@ To start using your cluster, you need to run the following as a regular user:
   sudo cp -i %s $HOME/.kube/config
   sudo chown $(id -u):$(id -g) $HOME/.kube/config
   kubectl get nodes
+`, botanist.PathKubeconfig)
+
+	for _, worker := range b.Shoot.GetInfo().Spec.Provider.Workers {
+		if worker.ControlPlane == nil {
```

Comment:
What is this check supposed to do? Is this just to check that there exists a worker group, which is not for control plane nodes? Would a simple check of the workers length be sufficient?
Not sure that I get the point.

Reply:
This check is to make sure the info about how to join worker nodes is only displayed if there is at least one non-control plane worker pool. Checking the workers length is not sufficient - if you only have one pool with `controlPlane: {}` then the message does not need to be printed.

In the long-run, this will change anyways, I guess, due to the fact that we will be able to join control plane nodes as well. So it's just a temporary state.